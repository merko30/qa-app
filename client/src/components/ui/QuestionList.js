import React, { PureComponent } from "react";

import Question from "./Question";

export default class QuestionList extends PureComponent {
  render() {
    const { questions, editQuestion, removeQuestion } = this.props;
    return (
      <div className="row mt-5">
        {questions &&
          questions.map(q => {
            return (
              <Question
                question={q}
                key={q.id}
                editQuestion={editQuestion}
                removeQuestion={removeQuestion}
              />
            );
          })}
      </div>
    );
  }
}
