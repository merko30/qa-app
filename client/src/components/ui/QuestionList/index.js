import React, { Component } from "react";
import PropTypes from "prop-types";

import Question from "../Question";

export default class QuestionList extends Component {
  static propTypes = {
    questions: PropTypes.array.isRequired,
    editQuestion: PropTypes.func.isRequired,
    removeQuestion: PropTypes.func.isRequired,
    loggedIn: PropTypes.bool.isRequired
  };

  render() {
    const {
      questions,
      editQuestion,
      removeQuestion,
      loggedIn,
      searchByTag
    } = this.props;
    return (
      <div className="row mt-5">
        {questions &&
          questions.map(q => {
            return (
              <Question
                question={q}
                key={q.id}
                editQuestion={editQuestion}
                removeQuestion={removeQuestion}
                loggedIn={loggedIn}
                searchByTag={searchByTag}
              />
            );
          })}
      </div>
    );
  }
}
