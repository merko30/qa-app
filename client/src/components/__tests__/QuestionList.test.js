import React from "react";

import renderWithRouter from "../../utils/renderWithRouter";

import QuestionList from "../ui/QuestionList";

it("renders the component and matches the snapshot", () => {
  const props = {
    questions: [
      {
        text: "question1",
        id: 1,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      },
      {
        text: "question2",
        id: 2,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      },
      {
        text: "question3",
        id: 3,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      }
    ]
  };

  const container = renderWithRouter(<QuestionList {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});

it("renders the right number of questions", () => {
  const props = {
    questions: [
      {
        text: "question1",
        id: 1,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      },
      {
        text: "question2",
        id: 2,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      },
      {
        text: "question3",
        id: 3,
        user: { name: "John Doe" },
        createdAt: Date.now(),
        answers: []
      }
    ]
  };

  const { getAllByText } = renderWithRouter(<QuestionList {...props} />);

  const questions = getAllByText(/question/i);

  expect(questions.length).toEqual(3);
});
