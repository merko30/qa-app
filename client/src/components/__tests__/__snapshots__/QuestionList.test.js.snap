// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component and matches the snapshot 1`] = `undefined`;
